### DISCUSSION TITLE
v0.7.0 Preview1

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#164] [Announcements]
这是一个技术预览版，只支持 FSR 缩放。

更改：
1. 显著降低 GPU 资源占用
2. 支持更多分辨率
3. 添加两种新的捕获模式

系统需求：
1. 任意版本的 Windows 10/11，但 Graphics Capture (WinRT Capture) 只在 v1903 及更新版本中可用
2. DirectX 11.1

其他更改：
在主界面可以调整 FSR 锐度（仅限此版本）

关于此版本的反馈请发布在 Discussion 中

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.7.0_preview1'>v0.7.0 Preview1</a>.</em>

**Comment by ZTexYmyr** [comment]
这版本可用于多人游戏吗?
想知道会不会对游戏主要挡案进行修改导致封号等

**Reply by palxex** [reply]
早期的Magpie可能算有，有注入动作以防止双光标，但也是可选的。
0.6.1以后作者更改了实现，现在就相当于一个放大到全屏的放大镜，原游戏进程不再有可能会受到任何直接的侵入。但网络游戏的保护系统千奇百怪，很多都会扫描全部系统进程，谁也说不准有没有采取宁可错杀一千策略的，这个包票没人能打。只能说正常思路的应该是没问题。

**Reply by ZTexYmyr** [reply]
了解，感谢解惑

**Reply by neavo** [reply]
如何重置设置到默认设置呢，在程序路径下好像没看到储存配置的文件之类的东西

**Reply by Blinue** [reply]
作为一款游戏工具，Magpie（v0.6.1 及更新版本） 对游戏是无侵入的，而且 Magpie 不注入，不修改内核，不会读写进程以外的内存，所调用的系统 API 都相当常用且安全，因此基本没有被封号的可能。如果被误封，申诉也相当容易，因为 Magpie 是完全的自由软件。但我无法提供保证，使用本工具的后果应由你自己承担。

**Reply by Blinue** [reply]
目前不能恢复默认设置。

Magpie 的设置文件一般保存在 C:\Users\\{USER\}\AppData\Local\Magpie

**Reply by neavo** [reply]
> 目前不能恢复默认设置。
> 
> Magpie 的设置文件一般保存在 C:\Users\{USER}\AppData\Local\Magpie

收到，放在程序目录下做成完全的便携版会不会更好一点？

**Reply by Blinue** [reply]
设置文件是.NET运行时自动管理的。所有配置都可以通过UI更改，直接修改设置文件也没有必要。

**Reply by neavo** [reply]
> 设置文件是.NET运行时自动管理的。所有配置都可以通过UI更改，直接修改设置文件也没有必要。

但是这样的话，改动设置的时候，每台设备都要设置一次，对设备比较对的用户来说有点麻烦 =。=

**Comment by MorderczkazPolski12** [comment]
This is a technical preview version and only supports FSR scaling.

Changes.

Significant reduction in GPU resource usage
Support for more resolutions
Adds two new capture modes
System requirements.

Any version of Windows 10/11, but Graphics Capture (WinRT Capture) is only available in v1903 and later
DirectX 11.1
Other changes.
FSR sharpness can be adjusted in the main screen (this version only)

Feedback on this version can be posted in Discussion

Translated with www.DeepL.com/Translator (free version)

Did the translator, translated it correctly? and is there any difference in FPS gain between 5.2 and 7.1? using FSR?
and what's the most efficient capture method for FSR? ( that takes the little amount of resources to give the most performance )

**Reply by Blinue** [reply]
You can trust DeepL. The GPU load of v0.7.0 is much less, but the FPS gain is hardware dependent.

**Reply by MorderczkazPolski12** [reply]
ok thanks alot.

I kind of trust DeepL when translating in Polish to English and vice versa, but in some languages, it completely didn't care about grammar, or completely changed the sentence / had same word twice even if it had different meaning, but not gonna question AI... )

**Comment by yutouyes** [comment]
2021-10-02 19:19:15.177|info|App.cpp:App::Initialize|正在初始化 App
2021-10-02 19:19:15.177|info|App.cpp:App::Initialize|已初始化 COM
2021-10-02 19:19:15.177|info|App.cpp:App::_RegisterHostWndClass|已注册主窗口类
2021-10-02 19:19:15.177|info|App.cpp:App::Initialize|源窗口客户区尺寸：1600x900
2021-10-02 19:19:15.181|info|App.cpp:App::_CreateHostWnd|主窗口尺寸：2560x1440
2021-10-02 19:19:15.181|info|App.cpp:App::_CreateHostWnd|已创建主窗口
2021-10-02 19:19:15.231|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
2021-10-02 19:19:15.231|critical|App.cpp:App::Initialize|初始化 Renderer 失败，正在清理
2021-10-02 19:19:15.233|info|App.cpp:App::Run|开始接收窗口消息
2021-10-02 19:19:15.233|info|CursorRenderer.cpp:CursorRenderer::~CursorRenderer|CursorRenderer 已析构
2021-10-02 19:19:15.233|info|App.cpp:App::Run|主窗口已销毁
2021-10-02 19:19:15.233|info|DllMain.cpp:Run|App 初始化失败，返回 GENREIC 消息


缩放失败，报错日志如上

环境是ryzen 3600+ GTX titan black，windows 10 21H1

**Reply by Blinue** [reply]
如果硬件不支持Feature Level 11.1可以使用下面的版本
[Magpie.zip](https://github.com/Blinue/Magpie/files/7244125/Magpie.zip)

**Reply by Nerboruto** [reply]
Does 11.1 support have any advantage over 11.0?

**Reply by yutouyes** [reply]
> 如果硬件不支持Feature Level 11.1可以使用下面的版本 [Magpie.zip](https://github.com/Blinue/Magpie/files/7244125/Magpie.zip)

![image](https://user-images.githubusercontent.com/17039457/135715087-040335fc-e37a-4dcb-81dc-b4d7564e6a7a.png)
Feature Level 11.1是指Direct3D功能嘛

**Reply by Blinue** [reply]
> Does 11.1 support have any advantage over 11.0?

This allows me to access more hardware features for better performance. Feature Level 11.1 will not be required in the release version.

> Feature Level 11.1是指Direct3D功能嘛

是的

**Comment by xing0999** [comment]
FL11.1只有AMD GCN以及Nvidia Maxwell以及之后的显卡才支持呀

老的fermi kepler以及amd 6000 5000系显卡都只支持FL11.0，这些老卡应该也是很需要FSR助力的

**Reply by yutouyes** [reply]
> FL11.1只有AMD GCN以及Nvidia Maxwell以及之后的显卡才支持呀
> 
> 老的fermi kepler以及amd 6000 5000系显卡都只支持FL11.0，这些老卡应该也是很需要FSR助力的



> > Does 11.1 support have any advantage over 11.0?
> 
> This allows me to access more hardware features for better performance. Feature Level 11.1 will not be required in the release version.
> 




正式版不会要求11.1了

**Comment by Apprisco** [comment]
Very laggy on 0.7.0 preview compared to 0.6.1, any ideas?

**Reply by Blinue** [reply]
Which game and which capture method?

**Reply by Apprisco** [reply]
Minecraft, GraphicsCapture aka the old RT.
Using pascal Gtx 10s, wondering if there's a reason why fineline asked.

**Comment by FineLine0** [comment]
is there a gpu Load difference between pascal (GTX 10's) card vs turing/ampere (RTX) card?

**Reply by Blinue** [reply]
Magpie is not designed for a specific architecture.

**Comment by FineLine0** [comment]
any plans on making more scalers? i would love a bicubic scaler and many other scalers! please and thank you!
look at these pages
https://newbedev.com/which-resize-algorithm-to-choose-for-videos
https://ijarcce.com/wp-content/uploads/2016/02/IJARCCE-7.pdf

**Reply by Blinue** [reply]
Please use v0.6.1, which supports many advanced scaling algorithms.

**Comment by cheng1nuo** [comment]
我发现放大后的窗口会锁 60 FPS （开启游戏内的显示帧率显示游戏内渲染帧率大于 60 fps 但是实际的帧率只有 60 fps )  后期会考虑移除锁定 60fps 吗

**Reply by Blinue** [reply]
GraphicsCapture之外的捕获模式的帧率取决于垂直同步，所有的捕获模式都是锁60fps吗

**Reply by cheng1nuo** [reply]
使用的是 版本是 Magpie_v0.7.0_preview 发现 所有的捕获模式都是 60FPS  （测试 游戏 League of Legends 右上角显示 200 FPS 开启 显示帧率 左上角的帧率是 60fps

**Reply by Blinue** [reply]
你的显示器刷新率是多少，支持动态刷新率吗

**Reply by cheng1nuo** [reply]
显示器型号为 U2790B 4k@60hz 10位色 不支持动态刷新率（FreeSync,G-Sync)
显卡为 AMD RX 580

**Reply by Blinue** [reply]
即使游戏内刷新率为200FPS，显示器也无法显示。建议你在游戏内开启垂直同步，可以节省系统资源。

**Reply by cheng1nuo** [reply]
但是 垂直同步这种方法可能会导致帧生成时间提高。

**Reply by Blinue** [reply]
所有窗口化的游戏都是强制垂直同步的，虽然游戏本身以很高的帧率运行，它在屏幕上的显示始终为60FPS。所以高于60FPS是没有意义的，除非游戏支持“关闭垂直同步”，在不支持可变帧率的显示器上这会导致画面撕裂。

这是一个支持关闭垂直同步的版本，你可以尝试一下
[Magpie.zip](https://github.com/Blinue/Magpie/files/7321950/Magpie.zip)

GraphicsCapture 依然是 60 FPS，这是 API 的限制。

**Reply by hooke007** [reply]
dwm的垂直是否和游戏全屏模式的垂直不同，独占全屏开游戏内垂直的情况偶尔能感知，为了避免渲染撕裂我都是无边框全屏，奇怪的是迄今为止我没有感受到dwm垂直带来的延迟问题

**Reply by Blinue** [reply]
画面撕裂应该是游戏本身的问题。在支持DirectFlip的硬件上无边框全屏不会引入额外的延迟

**Reply by palxex** [reply]
@Blinue 问一下是否有支持directflip的硬件列表？跟wddm版本直接相关吗

**Reply by Blinue** [reply]
查看日志文件里的“Multiplane Overlay 支持”和“Hardware Composition 支持”。或者使用[PresentMon](https://github.com/GameTechDev/PresentMon)查看Magpie的PresentMode，在支持directflip的硬件上应该是“Hardware Composed: Independent Flip”，这种情况下延迟应小于16.6ms

**Reply by palxex** [reply]
> 查看日志文件里的“Multiplane Overlay 支持”和“Hardware Composition 支持”。或者使用[PresentMon](https://github.com/GameTechDev/PresentMon)查看Magpie的PresentMode，在支持directflip的硬件上应该是“Hardware Composed: Independent Flip”，这种情况下延迟应小于16.6ms

嗯嗯，我机器上这俩都是否，所以才想找找硬件列表看什么显卡才能支持。

**Comment by ghost** [comment]
Hello, I just want to know what is this dwmsharedsurface and gdiovergxdi? and does it work better than the graphics capture?


你好，我只想知道这个dwmsharedsurface和gdiovergxdi是什么？它是否比图形捕获更好？

**Reply by Blinue** [reply]
GraphicsCapture is the most recommended capture method. You can try other methods if you encounter a CPU bottleneck.

**Comment by FranTesla** [comment]
GDI over DXGI feels a lot more responsive that other methods on a 165hz monitor (GTX1080Ti Windows 11)
(but it stutters in some heavy demanding games, in that case DwmSharedSurface seems best)
just feedback

**Reply by Blinue** [reply]
thanks :)

**Comment by yutouyes** [comment]
反馈一个笔记本的情况：
游戏是原神，运行于外接的2k显示器上，显示器连接在独显上
![image](https://user-images.githubusercontent.com/17039457/138838354-a2bed9a4-7d73-46f5-ae5c-761eb4087376.png)
关闭magpie时，核显占用在10%内，打开magpie后会上升至50%左右
![image](https://user-images.githubusercontent.com/17039457/138839167-adfa127c-cf00-49cc-944d-d17837f1f321.png)

设备：
CPU : AMD Ryzen  5 4600H，核显为vega 6
GPU : Nvidia RTX 2060 Mobile Refresh
系统是 WIndows 10 2004

**Reply by Blinue** [reply]
Magpie 运行在核显还是独显上？在核显上的话这是正常情况

**Reply by yutouyes** [reply]
> Magpie 运行在核显还是独显上？在核显上的话这是正常情况
感谢，已经解决

**Comment by gabor0928** [comment]
In Far Cry Primal, i have this issue in every windowed mode. When i click with any of the mouse buttons the game window appears immediately like PIP mode while the scaled remains in the background in full screen. The game runs in both of the windows at the same time... What can i do?

![image](https://user-images.githubusercontent.com/93269579/139083748-4cc11186-43f8-432d-b837-802494e48a10.png)

**Reply by Blinue** [reply]
Magpie doesn't work on windows that are top-most. I think the game has an option to turn this off.

**Reply by gabor0928** [reply]
Thanks for the answer you were right about the cause of the issue. I found the solution:

I needed to download a tool called "zero dump" to modify properties of the Far Cry window then set the "Disabled" option on. After this magpie works without window issue! :)

![farcry](https://user-images.githubusercontent.com/93269579/139109589-be7372e8-87cf-4c65-9049-af2f96835e2e.jpg)

**Reply by ghost** [reply]
@gabor0928 hey where can I find that tool ? can you share the link ?

**Reply by gabor0928** [reply]
> @gabor0928 hey where can I find that tool ? can you share the link ?

Here:
https://www.softpedia.com/get/Programming/Debuggers-Decompilers-Dissasemblers/zero-Dump.shtml

**Reply by ghost** [reply]
@gabor0928 hey thanks for the link.. i tried this for forza horizon 4 but still doesnt work.. check newest post in this conversation you will see a screenshot what the problem is

**Reply by gabor0928** [reply]
> @gabor0928 hey thanks for the link.. i tried this for forza horizon 4 but still doesnt work.. check newest post in this conversation you will see a screenshot what the problem is

I used it for Far Cry Primal only. I don't know if it's working with other titles.

**Comment by MorderczkazPolski12** [comment]
Hello.

Recently I've encountered problem out of the blue, one day Magpie + MSI Afterburner (with RTSS ) ( overlay statistics ) were working fine, The next day I wanted to play, turn of magpie, MSI Afterburner and game ( same issues across all games ) and MSI Afterburner statistics went crazy, tried playing with different settings in RiverTunerStatisticServer but no use, updating MSI+RTSS gave more troubles than fixing it, Capture method is and was set to "GameCapture" but this happens in all Capture Method's, any


Before the glitch/bug happen ( all was working fine as seen on screenshot, resolution was 1440x900 if I recall correctly )
![image_2021-10-31_165403](https://user-images.githubusercontent.com/88550829/139594005-dcd272e2-f861-42b7-83aa-6cfd11159fa0.png)

after the glitch happen ( 1280x800 at the time of taking the screenshot, but it happens on all resolutions )
![image_2021-10-31_165536](https://user-images.githubusercontent.com/88550829/139594050-d7eee1b1-33aa-459d-8110-e476433c96c5.png)


I've Looked for Saint'r Row 3 Remaster Screenshot to compare directly, but completed the game 100% and deleted straight afterwards, before the glitch happened,

**Reply by palxex** [reply]
It's not glitch, but was caused by magpie implement changes. Before 0.7 magpie was a direct2d window, which will not be captured by rtss, so afterburner will only show on original window, and scaled in magpie window. But in 0.7 magpie window rendering in d3d, that means itself will also owe a RTSS overlay, and it may displays on top of the baked-in rtss overlay of the original game window.
As a workaround, setup a profile for magpie.exe in rtss, and disable overlay for it.

**Reply by MorderczkazPolski12** [reply]
Ok, thanks, will give it a try!

**Comment by Coliostro** [comment]
Hi, dude.
What do you think about possibility to create a MagPie for DirectX 10/10.1 videocards? Theoretically is it possible? May be FSR can works through the OpenGL, like on Linux...
We all live in the era of mining, videocards for $1000. We have used video cards from eBay, or new GT 210 GT 220. They do not support DX 11.1. Personally, I have Ati Radeon HD 3850 1 GB DXlevel 10.1, I think FSR would be like a second wind for all old videocards. 

2021-11-03 19:45:28.047|critical|App.cpp:App::Initialize|初始化 Renderer 失败，正在清理
2021-11-03 19:45:28.053|info|App.cpp:App::Run|开始接收窗口消息
2021-11-03 19:45:28.053|info|CursorRenderer.cpp:CursorRenderer::~CursorRenderer|CursorRenderer 已析构
2021-11-03 19:45:28.053|info|App.cpp:App::Run|主窗口已销毁
2021-11-03 19:45:28.053|info|DllMain.cpp:Run|App 初始化失败，返回 GENREIC 消息
2021-11-03 19:45:39.834|info|App.cpp:App::Initialize|正在初始化 App
2021-11-03 19:45:39.834|info|App.cpp:App::Initialize|源窗口客户区尺寸：1280x720
2021-11-03 19:45:39.841|info|App.cpp:App::_CreateHostWnd|主窗口尺寸：1920x1080
2021-11-03 19:45:39.841|info|App.cpp:App::_CreateHostWnd|已创建主窗口
2021-11-03 19:45:39.843|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
![wwerwerwer](https://user-images.githubusercontent.com/93672460/140140096-9d1010e1-2c00-4972-bfc4-7d3d68f21e6f.PNG)
04

**Reply by hooke007** [reply]
https://github.com/Blinue/Magpie/discussions/164#discussioncomment-1418157

**Reply by Coliostro** [reply]
Doesn't work...
For all methods:
"2021-11-03 20:34:29.550|info|App.cpp:App::Initialize|正在初始化 App
2021-11-03 20:34:29.550|info|App.cpp:App::Initialize|已初始化 COM
2021-11-03 20:34:29.550|info|App.cpp:App::_RegisterHostWndClass|已注册主窗口类
2021-11-03 20:34:29.551|info|App.cpp:App::Initialize|源窗口客户区尺寸：440x201
2021-11-03 20:34:29.556|info|App.cpp:App::_CreateHostWnd|主窗口尺寸：1920x1080
2021-11-03 20:34:29.556|info|App.cpp:App::_CreateHostWnd|已创建主窗口
2021-11-03 20:34:29.564|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
2021-11-03 20:34:29.565|critical|App.cpp:App::Initialize|初始化 Renderer 失败，正在清理
2021-11-03 20:34:29.567|info|App.cpp:App::Run|开始接收窗口消息
2021-11-03 20:34:29.567|info|CursorRenderer.cpp:CursorRenderer::~CursorRenderer|CursorRenderer 已析构
2021-11-03 20:34:29.567|info|App.cpp:App::Run|主窗口已销毁
2021-11-03 20:34:29.567|info|DllMain.cpp:Run|App 初始化失败，返回 GENREIC 消息
2021-11-03 20:35:10.285|info|App.cpp:App::Initialize|正在初始化 App
2021-11-03 20:35:10.285|info|App.cpp:App::Initialize|源窗口客户区尺寸：1280x720
2021-11-03 20:35:10.294|info|App.cpp:App::_CreateHostWnd|主窗口尺寸：1920x1080
2021-11-03 20:35:10.294|info|App.cpp:App::_CreateHostWnd|已创建主窗口
2021-11-03 20:35:10.296|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
2021-11-03 20:35:10.296|critical|App.cpp:App::Initialize|初始化 Renderer 失败，正在清理
2021-11-03 20:35:10.302|info|App.cpp:App::Run|开始接收窗口消息
2021-11-03 20:35:10.302|info|CursorRenderer.cpp:CursorRenderer::~CursorRenderer|CursorRenderer 已析构
2021-11-03 20:35:10.302|info|App.cpp:App::Run|主窗口已销毁
2021-11-03 20:35:10.302|info|DllMain.cpp:Run|App 初始化失败，返回 GENREIC 消息
2021-11-03 20:35:24.179|info|App.cpp:App::Initialize|正在初始化 App
2021-11-03 20:35:24.179|info|App.cpp:App::Initialize|源窗口客户区尺寸：347x201
2021-11-03 20:35:24.186|info|App.cpp:App::_CreateHostWnd|主窗口尺寸：1920x1080
2021-11-03 20:35:24.186|info|App.cpp:App::_CreateHostWnd|已创建主窗口
2021-11-03 20:35:24.189|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
2021-11-03 20:35:24.189|critical|App.cpp:App::Initialize|初始化 Renderer 失败，正在清理
2021-11-03 20:35:24.195|info|App.cpp:App::Run|开始接收窗口消息
2021-11-03 20:35:24.195|info|CursorRenderer.cpp:CursorRenderer::~CursorRenderer|CursorRenderer 已析构
2021-11-03 20:35:24.195|info|App.cpp:App::Run|主窗口已销毁
2021-11-03 20:35:24.195|info|DllMain.cpp:Run|App 初始化失败，返回 GENREIC 消息
2021-11-03 20:35:34.109|info|App.cpp:App::Initialize|正在初始化 App
2021-11-03 20:35:34.109|info|App.cpp:App::Initialize|源窗口客户区尺寸：347x201
2021-11-03 20:35:34.116|info|App.cpp:App::_CreateHostWnd|主窗口尺寸：1920x1080
2021-11-03 20:35:34.116|info|App.cpp:App::_CreateHostWnd|已创建主窗口
2021-11-03 20:35:34.118|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
2021-11-03 20:35:34.119|critical|App.cpp:App::Initialize|初始化 Renderer 失败，正在清理
2021-11-03 20:35:34.124|info|App.cpp:App::Run|开始接收窗口消息
2021-11-03 20:35:34.124|info|CursorRenderer.cpp:CursorRenderer::~CursorRenderer|CursorRenderer 已析构
2021-11-03 20:35:34.124|info|App.cpp:App::Run|主窗口已销毁
2021-11-03 20:35:34.124|info|DllMain.cpp:Run|App 初始化失败，返回 GENREIC 消息
"
![цуцкцукцук](https://user-images.githubusercontent.com/93672460/140162397-a7819f23-4b88-4678-a576-55991ac655a2.PNG)

**Reply by Blinue** [reply]
You should check your drivers #195 

https://github.com/Blinue/Magpie/discussions/164#discussioncomment-1418157 
This build requires feature level 10.0, which should be supported by your hardware accroding to GPU-Z

**Reply by Coliostro** [reply]
It is last official Ati Radeon HD 3XXX legacy driver... 
Log still shows: "2021-11-10 21:36:14.663|critical|Renderer.cpp:Renderer::_InitD3D|**D3D11CreateDevice** 失败
	HRESULT：0x887A0004" 
Still doesn't works... Guys, does anyone have a DirectX10 graphics card where Magpie works correctly?

**Comment by yutouyes** [comment]
请问N卡带的shadowplay怎么设置才可以录制放大之后的画面，比如说游戏分辨率1600x900，magpie放大到全屏2k1440p，shadowplay录制的画面分辨率还是1600x900的，我想录制放大之后的画面效果，这个要怎么做呢

**Reply by yutouyes** [reply]
尝试过直接录制桌面，在alt+f11放大游戏窗口至全屏后，录像里就只有放大前最后一刻的静止图像，放大之后的游戏画面无法录制

**Reply by Blinue** [reply]
我没使用过shadowplay，如果它不支持录制 Magpie 你可能需要更换录制软件

**Reply by yutouyes** [reply]
> 我没使用过shadowplay，如果它不支持录制 Magpie 你可能需要更换录制软件

那请问你用的是哪个，我换了obs也是类似情况

**Reply by Blinue** [reply]
在OBS里捕获整个显示器

**Reply by yutouyes** [reply]
> 在OBS里捕获整个显示器

我也是捕获的显示器，还是只能录到静止的画面。用的编码器是显卡的。
也尝试过选择捕获magpie窗口，这个也录不到正常的游戏画面

**Reply by yutouyes** [reply]
> > 在OBS里捕获整个显示器
> 
> 我也是捕获的显示器，还是只能录到静止的画面。用的编码器是显卡的。 也尝试过选择捕获magpie窗口，这个也录不到正常的游戏画面

用CPU的x264编码器可以正常录制画面了，但是效率比显卡编码器低，最后帧数不高，反应不出magpie的提升QAQ

**Reply by Blinue** [reply]
![image](https://user-images.githubusercontent.com/34770031/140634373-23308b3f-31f9-43aa-addf-26d5de5971a6.png)
尝试更换捕获方式。如果是硬件差别我也没法帮你

**Reply by yutouyes** [reply]
> ![image](https://user-images.githubusercontent.com/34770031/140634373-23308b3f-31f9-43aa-addf-26d5de5971a6.png) 尝试更换捕获方式。如果是硬件差别我也没法帮你

是这个方式。你的编码器选择的是显卡吗
![image](https://user-images.githubusercontent.com/17039457/140638277-d2e4a68e-f6b2-4ae7-a935-bc5ef89d9050.png)

**Reply by Blinue** [reply]
每一种编码器我都测试了，都可以捕获到

**Reply by yutouyes** [reply]
> 每一种编码器我都测试了，都可以捕获到

感谢，不是magpie的问题，是我obs没有给管理员权限的原因

**Comment by neavo** [comment]
反馈BUG：在设置中取消勾选“显示帧率”，实质上取消的是“开机启动时自动最小化到系统托盘”功能

**Reply by Blinue** [reply]
能否详细描述一下，我没法复现这个bug

**Reply by neavo** [reply]
> 能否详细描述一下，我没法复现这个bug

奇怪，我也复现不了了 ... 等我研究出怎么稳定复现了再来回复

**Comment by biantaonkss** [comment]
大佬您好，我下载了您的0.7.0_preview1版本的Magpie，发现在有些游戏里当FSR锐度拉到最右边时画面会有很多像素点，并且左上角会出现两个afterburner，帧数也比较低
![1](https://user-images.githubusercontent.com/93912134/140725429-7b741529-5be2-4c2e-a882-d8f87dd87d3c.png)
![2](https://user-images.githubusercontent.com/93912134/140725461-aab555e1-8e72-449f-a49f-91a27c2333ef.png)
这两张是帝国时代2决定版和崩坏3的截图，但我在玩Minecraft的时候用同样的方法却没遇到这种情况，帧数得到了大幅提升，我手里只有这几款游戏，希望能有所帮助，请问这种问题是什么原因引起的呢？

**Reply by Blinue** [reply]
1. 产生噪点是因为锐化过度，建议不要将FSR锐度调的太高
2. 两个afterburner统计的问题可以参考 https://github.com/Blinue/Magpie/discussions/164#discussioncomment-1567168
3. 帧率低是受限于你的显卡性能吗？

**Reply by biantaonkss** [reply]
感谢大佬回复，我的显卡是核显（UHD630），比较差，但玩帝国时代2和崩坏3设置好画面选项可以八九十帧，开了FSR反倒掉帧严重，有肉眼可见的卡顿，Minecraft则会比原来的帧数高不少，很流畅

**Reply by Blinue** [reply]
性能问题可以参考这个链接 https://github.com/Blinue/Magpie/issues/94#issuecomment-907728719

[Magpie.zip](https://github.com/Blinue/Magpie/files/7446853/Magpie.zip) 这个版本支持关闭垂直同步，一般可以提高帧率

**Reply by biantaonkss** [reply]
好的，感谢大佬

**Reply by Nerboruto** [reply]
dx11.0 no longer supported?
2021-11-09 19:31:48.022|critical|Renderer.cpp:Renderer::_InitD3D|D3D11CreateDevice 失败
	HRESULT：0x887A0004
2021-11-09 19:31:48.022|critical|App.cpp:App::Run|初始化 Renderer 失败，正在清理

**Reply by Blinue** [reply]
https://github.com/Blinue/Magpie/discussions/164#discussioncomment-1418157

**Reply by Nerboruto** [reply]
the last one you posted doesn't work... the old one does. I wanted to know for the last one.

**Reply by Blinue** [reply]
Could you post the logs of the working version? They can help me to analyze the hardware support.

**Reply by Nerboruto** [reply]
[logs.zip](https://github.com/Blinue/Magpie/files/7520852/logs.zip)

done

kepler dx11.0 feature level

**Reply by Blinue** [reply]
Thank you

**Comment by ghost** [comment]
@Blinue  I have just found this tool yesterday and it works with most of the game titles perfectly like gta5, wd legions, rise of the tomb raider, re village etc.. I really thank you for this tool. I hope you continue to develop as expanding features.. but somehow I couldnt manage to get this tool work with Forza Horizon 4 perfectly.. after FSR applied game doesn`t properly scaled up as full screen.it is ms store version. here is the screenshot below. what can I do ? thanks again! [fh4](https://user-images.githubusercontent.com/57366698/141306690-793d6114-ff1d-471c-92a5-175322022ab1.png)

**Reply by Blinue** [reply]
Magpie doesn't support UWP so far, I have a plan though

**Reply by ghost** [reply]
ok understand now. I hope your plan works in future updates and i like the new sharpening ratio in your latest preview one. thanks
